tests/ui/transmute/redundant-transmutation.stderr

```diff
@@ -0,0 +1,235 @@
+error: this transmute could be performed safely
+  --> $DIR/redundant-transmutation.rs:7:14
+   |
+LL |     unsafe { transmute(x) }
+   |              ^^^^^^^^^^^^ help: replace `transmute`: `u32::to_ne_bytes(x)`
+   |
+   = help: there's also `to_le_bytes` and `to_ne_bytes` if you expect a particular byte order
```

Comment:
```suggestion
error: unnecessary transmute
  --> $DIR/redundant-transmutation.rs:7:14
   |
LL |     unsafe { transmute(x) }
   |              ^^^^^^^^^^^^ help: replace this with `u32::to_ne_bytes(x)`
   |
   = help: there's also `to_le_bytes` and `to_ne_bytes` if you expect a particular byte order
```
This currently looks a bit odd to me :thinking: like the error message doesn't quite match the *vibe* of our existing diagnostics